controllers/databaseclusterrestore_controller.go

```diff
@@ -246,7 +246,7 @@ func (r *DatabaseClusterRestoreReconciler) restorePSMDB(
 			}
 
 			psmdbCR.Spec.BackupSource = &psmdbv1.PerconaServerMongoDBBackupStatus{
-				Destination: fmt.Sprintf("s3://%s/%s", backupStorage.Spec.Bucket, restore.Spec.DataSource.BackupSource.Path),
+				Destination: restore.Spec.DataSource.BackupSource.Path,
```

Comment:
Does this change affect the "normal" restores to the same DB?

Reply:
Since this is only for the case when the backup source is used I guess not because we currently only use the backup name.

Reply:
P.S. feel free to resolve this conversation and merge the PR.

Reply:
Restore to the same db works, just checked. You're right, we never actually used the backup source for the restores, that's why I believe we didn't notice the weird way of constructing the path